src/upkeep.rs

```diff
@@ -273,24 +273,31 @@ pub async fn do_upkeep(
     metrics::histogram!("upkeep.duration").record(upkeep_start.elapsed());
 
     // Task statuses
-    metrics::counter!("upkeep.completed").increment(result_context.completed);
-    metrics::counter!("upkeep.failed").increment(result_context.failed);
-    metrics::counter!("upkeep.retried").increment(result_context.retried);
-    metrics::counter!("upkeep.expired").increment(result_context.expired);
+    metrics::counter!("upkeep.task.state_transition", "state" => "completed")
+        .increment(result_context.completed);
+    metrics::counter!("upkeep.task.state_transition", "state" => "failed")
+        .increment(result_context.failed);
+    metrics::counter!("upkeep.task.state_transition", "state" => "retried")
+        .increment(result_context.retried);
+    metrics::counter!("upkeep.task.state_transition", "state" => "expired")
+        .increment(result_context.expired);
 
     // Upkeep cleanup actions
-    metrics::counter!("upkeep.deadlettered").increment(result_context.deadlettered);
-    metrics::counter!("upkeep.discarded").increment(result_context.discarded);
-    metrics::counter!("upkeep.processing_attempts_exceeded")
+    metrics::counter!("upkeep.cleanup_action", "kind" => "deadlettered")
+        .increment(result_context.deadlettered);
+    metrics::counter!("upkeep.cleanup_action", "kind" => "discarded")
+        .increment(result_context.discarded);
+    metrics::counter!("upkeep.cleanup_action", "kind" => "processing_attempts_exceeded")
         .increment(result_context.processing_attempts_exceeded);
-    metrics::counter!("upkeep.processing_deadline_reset")
+    metrics::counter!("upkeep.cleanup_action", "kind" => "processing_deadline_reset")
         .increment(result_context.processing_deadline_reset);
-    metrics::counter!("upkeep.delay_elapsed").increment(result_context.delay_elapsed);
+    metrics::counter!("upkeep.cleanup_action", "kind" => "delay_elapsed")
+        .increment(result_context.delay_elapsed);
 
     // State of inflight tasks
-    metrics::gauge!("upkeep.pending_count").set(result_context.pending);
-    metrics::gauge!("upkeep.processing_count").set(result_context.processing);
-    metrics::gauge!("upkeep.delay_count").set(result_context.delay);
+    metrics::gauge!("upkeep.cur_pending_tasks").set(result_context.pending);
+    metrics::gauge!("upkeep.cur_processing_tasks").set(result_context.processing);
+    metrics::gauge!("upkeep.cur_delayed_tasks").set(result_context.delay);
```

Comment:
```suggestion
    metrics::gauge!("upkeep.current_pending_tasks").set(result_context.pending);
    metrics::gauge!("upkeep.current_processing_tasks").set(result_context.processing);
    metrics::gauge!("upkeep.current_delayed_tasks").set(result_context.delay);
```

We should use the full word imo.